plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,461 @@
+                isPartitionKey ? PARTITION_KEY : REGULAR,
+                Optional.empty());
+    }
+
+    private static class MissingColumnMetricsException
+            extends Exception
+    {}
+
+    private static SqlRow getColumnMetricBlock(RowType columnMetricType, Object min, Object max, Long nullCount)
+    {
+        return buildRowValue(columnMetricType, fieldBuilders -> {
+            List<RowType.Field> fields = columnMetricType.getFields();
+            writeNativeValue(fields.get(0).getType(), fieldBuilders.get(0), min);
+            writeNativeValue(fields.get(1).getType(), fieldBuilders.get(1), max);
+            writeNativeValue(fields.get(2).getType(), fieldBuilders.get(2), nullCount);
+        });
+    }
+
+    private static Optional<RowType> getMetricsColumnType(List<DeltaLakeColumnHandle> columns)
+    {
+        List<RowType.Field> metricColumns = columns.stream()
+                .map(column -> RowType.field(
+                        column.getBaseColumnName(),
+                        RowType.from(ImmutableList.of(
+                                new RowType.Field(Optional.of("min"), column.getType()),
+                                new RowType.Field(Optional.of("max"), column.getType()),
+                                new RowType.Field(Optional.of("null_count"), BIGINT)))))
+                .collect(toImmutableList());
+        if (metricColumns.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(RowType.from(metricColumns));
+    }
+
+    private static boolean isPrimitiveType(Class<?> type)
+    {
+        return (type == long.class || type == double.class || type == boolean.class || type == byte.class || type == int.class);
+    }
+
+    private record DeltaLakePartitionStatistics(long fileCount, long size, Map<String, Object> minValues, Map<String, Object> maxValues, Map<String, Long> nullCounts)
+    {
+        public static class Builder
+        {
+            private final List<DeltaLakeColumnHandle> columns;
+            private final TypeManager typeManager;
+            private long fileCount;
+            private long size;
+            private final Map<String, ColumnStatistics> columnStatistics = new HashMap<>();
+            private final Map<String, Optional<Long>> nullCounts = new HashMap<>();
+
+            public Builder(List<DeltaLakeColumnHandle> columns, TypeManager typeManager)
+            {
+                this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns is null"));
+                this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            }
+
+            public void acceptAddFileEntry(AddFileEntry addFileEntry)
+            {
+                if (addFileEntry.getDeletionVector().isPresent()) {
+                    return;
+                }
+
+                fileCount++;
+                size += addFileEntry.getSize();
+
+                addFileEntry.getStats().ifPresent(stats -> {
```

Comment:
if the stats of a single file are not present, we can't say anything in `data` column for this row.

Reply:
gentle reminder in the partition.

Reply:
Resolved